webapp/channels/src/packages/mattermost-redux/src/actions/channels.ts

```diff
@@ -256,114 +256,36 @@ export function createGroupChannel(userIds: string[]): NewActionFuncAsync<Channe
 }
 
 export function patchChannel(channelId: string, patch: Partial<Channel>): NewActionFuncAsync<Channel> {
-    return async (dispatch, getState) => {
-        dispatch({type: ChannelTypes.UPDATE_CHANNEL_REQUEST, data: null});
-
-        let updated;
-        try {
-            updated = await Client4.patchChannel(channelId, patch);
-        } catch (error) {
-            forceLogoutIfNecessary(error, dispatch, getState);
-
-            dispatch({type: ChannelTypes.UPDATE_CHANNEL_FAILURE, error});
-            dispatch(logError(error));
-            return {error};
-        }
-        dispatch(batchActions([
-            {
-                type: ChannelTypes.RECEIVED_CHANNEL,
-                data: updated,
-            },
-            {
-                type: ChannelTypes.UPDATE_CHANNEL_SUCCESS,
-            },
-        ]));
-
-        return {data: updated};
-    };
-}
-
-export function updateChannel(channel: Channel): NewActionFuncAsync { // HARRISONTODO unused
-    return async (dispatch, getState) => {
-        dispatch({type: ChannelTypes.UPDATE_CHANNEL_REQUEST, data: null});
-
-        let updated;
-        try {
-            updated = await Client4.updateChannel(channel);
-        } catch (error) {
-            forceLogoutIfNecessary(error, dispatch, getState);
-
-            dispatch({type: ChannelTypes.UPDATE_CHANNEL_FAILURE, error});
-            dispatch(logError(error));
-            return {error};
-        }
-
-        dispatch(batchActions([
-            {
-                type: ChannelTypes.RECEIVED_CHANNEL,
-                data: updated,
-            },
-            {
-                type: ChannelTypes.UPDATE_CHANNEL_SUCCESS,
-            },
-        ]));
-
-        return {data: updated};
-    };
+    return bindClientFunc({
+        clientFunc: Client4.patchChannel,
+        onSuccess: [ChannelTypes.RECEIVED_CHANNEL],
+        params: [channelId, patch],
+    });
```

Comment:
I see some behavior changes here, mainly a few events not being dispatched. Is that intended?

Reply:
Oh yeah, I removed the state stored in `state.requests.channels.update` because it wasn't used, so those actions weren't needed either. We've removed most of the request state by now, but we left a few pieces that are actually used. Since that one isn't used, I took this opportunity to remove it as well